src/hoc/enrich-analytics.tsx

```diff
@@ -1,46 +1,31 @@
 import * as React from 'react';
+import {ComponentType, FunctionComponent} from 'react';
 import {AnalyticsDispatcher} from 'shisell';
 import {wrapDisplayName} from '../wrapDisplayName';
-import {Requireable} from 'prop-types';
-
-import analyticsContextTypes, {AnalyticsContext} from '../analytics-context-types';
-import {withAnalytics} from './with-analytics';
-import {withoutAnalytics} from './without-analytics';
-import Analytics from '../analytics';
+import {ShisellContext} from '../shisell-context';
 
 type TransformAnalyticsFunc<T> = (dispatcher: AnalyticsDispatcher, props: T) => AnalyticsDispatcher;
-type DispatcherFactory = () => AnalyticsDispatcher;
-
-class LazyAnalytics {
-    constructor(private dispatcherFactory: DispatcherFactory) {}
 
-    get dispatcher() {
-        return this.dispatcherFactory();
-    }
+export function enrichAnalytics<Props>(transformAnalyticsFunc: TransformAnalyticsFunc<Props>) {
+    return (BaseComponent: ComponentType<Props>) => {
+        const EnhancedComponent: FunctionComponent<Props> = props => (
+            <ShisellContext.Consumer>
+                {analytics => (
+                    <ShisellContext.Provider
+                        value={{
```

Comment:
I'm not entirely sure, but won't this be victim to something like they describe [here](https://reactjs.org/docs/context.html#caveats), since we're recreating the object on every render?